lib/src/stream_group.dart

```diff
@@ -216,8 +229,17 @@ class StreamGroup<T> implements Sink<Stream<T>> {
   Future? _onCancel() {
     _state = _StreamGroupState.canceled;
 
-    var futures = _subscriptions.values
-        .map((subscription) => subscription!.cancel())
+    var futures = _subscriptions.entries
+        .map((entry) {
+          var subscription = entry.value;
+          if (subscription != null) return subscription.cancel();
```

Comment:
For total paranoia, consider moving this line inside the `try`. If the `cancel` call throws synchronously (then it's very badly behaved, but) it might cause other subscriptions to not get cancelled.